### PR TITLE
docs: updated okta oidc config

### PR DESCRIPTION
**Pull request by huynhj93**
### Summary

The OKTA OIDC docs are missing details on how to set up the OKTA application. I've added that and pictures to reference on how to do so. I've also added in detail how to edit the config maps and secrets for the client secret for the application. 

Checklist:

* [ c] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-cd/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this does not need to be in the release notes.
* [ n/a] The title of the PR states what changed and the related issues number (used for the release note).
* [n/a ] I've included "Closes [ISSUE #]" or "Fixes [ISSUE #]" in the description to automatically close the associated issue.
* [ n/a] I've updated both the CLI and UI to expose my feature, or I plan to submit a second PR with them.
* [ no] Does this PR require documentation updates?
* [ yes] I've updated documentation as required by this PR.
* [ no] Optional. My organization is added to USERS.md.
* [ yes] I have signed off all my commits as required by [DCO](https://github.com/argoproj/argoproj/tree/master/community#contributing-to-argo)
* [ n/a] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [ n/a] My build is green ([troubleshooting builds](https://argo-cd.readthedocs.io/en/latest/developer-guide/ci/)).

### CODE REVIEW

File: docs/operator-manual/user-management/okta.md [hunk 1]
```diff
@@ -116,36 +116,65 @@ data:
...
 !!! warning "Do you want groups for RBAC later?"
     If you want `groups` scope returned from Okta you need to unfortunately contact support to enable [API Access Management with Okta](https://developer.okta.com/docs/concepts/api-access-management/) or [_just use SAML above!_](#saml-with-dex)
 
     Next you may need the API Access Management feature, which the support team can enable for your OktaPreview domain for testing, to enable "custom scopes" and a separate endpoint to use instead of the "public" `/oauth2/v1/authorize` API Access Management endpoint. This might be a paid feature if you want OIDC unfortunately. The free alternative I found was SAML.
 
+# Set up your Authorization Server in OKTA
 1. On the `Okta Admin` page, navigate to the Okta API Management at `Security > API`.
     ![Okta API Management](../../assets/api-management.png)
-1. Choose your `default` authorization server.
+1. Choose your `default` authorization server, or create a new one called `argo`, which an audience of `api://argo`.
```
Comment (crenshaw-dev) [inline docs/operator-manual/user-management/okta.md:129]: s/which/with ?

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
This looks awesome! Hoping someone more familiar w/ Okta can also review. In the meantime, one nitpick.

**Comment by jdmaguire** [comment]
In terms of feedback, I was able to follow this and get everything working. Thanks for figuring this out!

File: docs/operator-manual/user-management/okta.md [hunk 2]
```diff
@@ -116,36 +116,65 @@ data:
   scopes: '[email,groups]'
 ```
 
-## OIDC (without Dex)
+## OIDC
```
Comment (jdmaguire) [inline docs/operator-manual/user-management/okta.md:119]: The link on line 9 needs to be updated.

Comment (john-kajabi) [reply docs/operator-manual/user-management/okta.md]: updated this line!

File: docs/operator-manual/user-management/okta.md [hunk 3]
```diff
@@ -116,36 +116,65 @@ data:
...
   issuer: https://yourorganization.oktapreview.com
   clientID: 0oaltaqg3oAIf2NOa0h3
-  clientSecret: ZXF_CfUc-rtwNfzFecGquzdeJ_MxM4sGc8pDT2Tg6t
+  clientSecret: $oidc.okta.clientSecret
   requestedScopes: ["openid", "profile", "email", "groups"]
   requestedIDTokenClaims: {"groups": {"essential": true}}
 ```
 <!-- markdownlint-enable MD046 -->
 
+1. Note, if you see `grpc client` errors, you might want to restart your argocd server: `kubectl rollout restart deploy/argocd-server -n argocd`
```
[the remark this answers is not in the document]
Comment (john-kajabi) [reply docs/operator-manual/user-management/okta.md]: also updated this!

File: docs/operator-manual/user-management/okta.md (hunk 2 above)
Comment (huynhj93) [reply docs/operator-manual/user-management/okta.md]: updated this line to match line 9

File: docs/operator-manual/user-management/okta.md (hunk 3 above)
[the remark this answers is not in the document]
Comment (huynhj93) [reply docs/operator-manual/user-management/okta.md]: updated!

File: docs/operator-manual/user-management/okta.md [hunk 4]
```diff
@@ -116,36 +116,65 @@ data:
...
+
+# Set up your Argo Application in OKTA
+1. On the `Okta Admin` page, navigate to the Applications at `Applications > Applications`.
+1. Click on `Create App Integration`.
+    1. Select `OIDC - OpenID connect` as your sign-in method.
+    1. Select `Web Application` as your application type.
+1. On the `New Web app integration` page, do the following:
+    1. Enter `ArgoCD` as the app integration name
+    1. Select `Authorization Code` and `Implicit(hybrid)` for `Grant Type`
+    1. For `Sign-in redirect URIs`, enter `https://${your.fqdn.com}/auth/callback` and `http://localhost:8085/auth/callback` (this is for signing in with CLI)
```
[the remark this answers is not in the document]
Comment (john-kajabi) [reply docs/operator-manual/user-management/okta.md:150]: left a note in line 152 that this isn't the correct url but the default value

**Review by todaywasawesome — CHANGES_REQUESTED** [review]
Closing in favor of changes made in #13811

**Pull request closed without merging**